packages/language-server/src/server.ts

```diff
@@ -98,32 +98,51 @@ connection.onCompletion(doAutoCompletion(documents, neo4jSchemaPoller));
 
 connection.onDidChangeConfiguration(
   (params: { settings: { neo4j: Neo4jSettings } }) => {
-    neo4jSchemaPoller.disconnect();
-
-    const neo4jConfig = params.settings.neo4j;
-    if (
-      neo4jSchemaPoller.connection === undefined &&
-      neo4jConfig.connect &&
-      neo4jConfig.password &&
-      neo4jConfig.connectURL &&
-      neo4jConfig.user
-    ) {
-      void neo4jSchemaPoller.persistentConnect(
-        neo4jConfig.connectURL,
-        {
-          username: neo4jConfig.user,
-          password: neo4jConfig.password,
-        },
-        { appName: 'cypher-language-server' },
-      );
-    }
+    changeConnection(params.settings.neo4j);
+  },
+);
+
+connection.onNotification(
```

Comment:
this receives notifications from the extension, presently we have a single method that it's listening for which signals that the connection has changed and we need to reconnect